### PR TITLE
Fix blank page when initiating a search with no search criteria

### PR DESCRIPTION
**Pull request by hamzawritescode**
## Summary

**Fix blank page when initiating a search with no search criteria**

## Ticket Link

Fixes https://github.com/mattermost/docs/issues/7031

### CODE REVIEW

**Comment by hamzawritescode** [comment]
_**Also, upon approval, where can I claim my free Mattermost mug?**_

File: source/_templates/search.html [hunk 1]
```diff
@@ -1,54 +1,70 @@
...
+	});
+
+	document.addEventListener('DOMContentLoaded', function() {
+	    const searchTitle = document.getElementById('search-title');
+	    const searchResults = document.getElementById('search-results');
+	    const searchSummary = document.getElementById('search-summary');
+	    const searchQuery = new URLSearchParams(window.location.search).get('q');
+
+	    if (!searchQuery || searchQuery.trim() === '') {
+	        searchTitle.textContent = "No search query entered!";
```
Comment (cwarnermm) [inline source/_templates/search.html:65]:
```suggestion
	        searchTitle.textContent = "Search Results";
```

File: source/_templates/search.html [hunk 2]
```diff
@@ -1,54 +1,70 @@
...
+	        searchSummary.textContent = "Please enter a search term to get results.";
```
Comment (cwarnermm) [inline source/_templates/search.html:66]:
```suggestion
	        searchSummary.textContent = "Please enter at least one search term.";
```

**Review by cwarnermm — APPROVED** [review]
Thanks, @hamzawritescode!

**Comment by hamzawritescode** [comment]
There you go, @cwarnermm! Also, where can I claim my mug?

**Comment by cwarnermm** [comment]
@hamzawritescode - Can you merge latest master into your fork please?

**Comment by hamzawritescode** [comment]
All done!

**Comment by varghesejose2020** [comment]
@lindy65  please assign me to test this PR

**Comment by lindy65** [comment]
Assigned you for QA testing @varghesejose2020 

You can test the changes using the preview environment [here](https://github.com/mattermost/docs/pull/7444#issuecomment-2397183679)

**Comment by varghesejose2020** [comment]
Screenshot_20241010_220915
 LGTM👍

Got one scenario if we type space on search bar and click enter `Please enter at least one search term.` this message is shown and at the end some results are shown on that page.Do both of these cases need to be displayed? Please refer the video

https://github.com/user-attachments/assets/8efca843-6ef1-4896-bafd-b8d5a35af703

**Comment by lindy65** [comment]
@cwarnermm - can you help with Varghese's question above please?

**Comment by cwarnermm** [comment]
Good question, @varghesejose2020! I consider this use case a pass. Thank you for identifying an edge case I hadn't considered!

**Comment by lindy65** [comment]
Thanks @cwarnermm :)

Thanks for testing @varghesejose2020 - counted towards Hacktoberfest!

**Pull request merged**